nservicebus/azure-service-bus/addressing-logic.md

```diff
@@ -0,0 +1,171 @@
+---
+title: Azure Service Bus Transport Addressing Logic
+summary: Azure ServiceBus Transport Addressing Logic
+tags:
+- Azure
+- Cloud
+- Configuration
+---
+
+## Physical Addressing Logic
+
+One of the responsibilities of the transport is determining the names and physical location of entities in the underlying physical infrastructure. This is achieved by turning logical endpoint names into physical addresses of the Azure Service Bus entities, which is called *Physical Addressing Logic*.
+
+Prior to version 7 there was no explicit layer for handling this kind of logic, most assumptions such as assumptions on length limitations, legal characters, differences between paths and names etc... were sprinkled across the transport.
+
+Over time, subtle variations started to show in these assumptions as changes took place inside the implementation of certain namespace types. For example `Mixed` namespaces allowed paths and names up 290 characters, while `Messaging` namespaces are capped at 260. Next to that, customers also figured out ways to game the system (for valid reasons), like embedding `'/'` characters in endpoint names to create subfolders in the Azure Service Bus namespace registration system.
+
```

Comment:
> Prior to version 7 there was no explicit layer for handling this kind of logic, most assumptions such as assumptions on length limitations, legal characters, differences between paths and names etc... were sprinkled across the transport.

> Over time, subtle variations started to show in these assumptions as changes took place inside the implementation of certain namespace types. For example `Mixed` namespaces allowed paths and names up 290 characters, while `Messaging` namespaces are capped at 260. Next to that, customers also figured out ways to game the system (for valid reasons), like embedding `'/'` characters in endpoint names to create subfolders in the Azure Service Bus namespace registration system.

In Versions 6 and below, the *Physical Addressing Logic* implementation made a number of implicit assumptions regarding names, such as length limitations, legal characters, differences between paths and names, etc. There was no way to explicitly control those assumptions, and over time subtle variations started to show up.

For example `Mixed` namespaces allowed paths and names to have up to 290 characters, while `Messaging` namespaces are capped at 260 characters. Moreover, customers were sometimes forced to game the system for valid reasons. For example they needed to figure out how to embed slashes (`'/'`) in endpoint names to create subfolders in the Azure Service Bus namespace registration system.